test/test_serialization.py

```diff
@@ -741,6 +741,16 @@ def test_pathlike_serialization(self):
             torch.save(model, path)
             torch.load(path)
 
+    def test_meta_serialization(self):
+        big_model = torch.nn.Conv2d(20000, 320000, kernel_size=3, device='meta')
+
+        with BytesIOContext() as f:
+            torch.save(big_model, f)
+            f.seek(0)
+            state = torch.load(f)
```

Comment:
What happens if I `map_location="cpu"`?
If it just crash and burn, maybe we want to add a check and raise a nice error?

Reply:
It actually silently just ignores the map location. I am guessing this is because mapping happens at storage processing, but it is bypassed. This seems like a legit bug that also affects mlc/xla serialization but I suggest we fix it in another PR.

Reply:
I'm happy with a fix in another PR if that also affect xla